### ISSUE TITLE
Test typing to avoid typing PRs

### ISSUE THREAD

**Issue by rom1504** [PrismarineJS/mineflayer#2519]
1. in all modules we will have a large test-typing.ts that just call everything and create everything (it will not work but it will compile)
2. we add tsc typings/test-typings in npm test of all modules
3. in all PRs adding a new thing in API, they will have to add something in test-typing.ts

example
https://github.com/PrismarineJS/node-minecraft-data/blob/master/package.json#L10
https://github.com/PrismarineJS/node-minecraft-data/blob/master/typings/test-typings.ts

**Comment by EnderTheCoder** [comment]
Hope this can work soon. I really want to use typescript to replace javascript in my project.

**Comment by rom1504** [comment]
This will not change a single thing with support of typescript. You can already use mineflayer in your typescript project

**Comment by EnderTheCoder** [comment]
It's just the problem of autocomplete. I just solved it.
When I tried to use mineflayer in the way of your example, vscode didn't give any suggestions.
 
![2022-03-22 10-16-10 的屏幕截图](https://user-images.githubusercontent.com/47713889/159394903-191e1c40-832e-44f5-879f-49310b863ee3.png)

But if I code something like "import mineflayer from ‘mineflayer’", it worked.

![2022-03-22 10-15-28 的屏幕截图](https://user-images.githubusercontent.com/47713889/159394926-a52504f1-11cc-47b4-8f2c-7461c3a38211.png)

**Comment by IceTank** [comment]
> But if I code something like "import mineflayer from ‘mineflayer’", it worked.

Yes because typescript dose not add types without `import` when using `require`. To get types with `require` you would need to do something like this `import mineflayer = require('mineflayer')`.
